patches/tModLoader/Terraria/ModLoader/IO/PlayerIO.cs

```diff
@@ -118,12 +118,23 @@ internal static class PlayerIO
 
 		public static void LoadResearch(Player player, IList<TagCompound> list) {
 			foreach (var tag in list) {
-				ContentSamples.ItemNetIdsByPersistentIds.TryGetValue(tag.GetString("persistentID"), out int netID);
-				ContentSamples.ItemsByType.TryGetValue(netID, out Item realItem);
-				if (ItemLoader.NeedsModSaving(realItem)) {
-					player.creativeTracker.ItemSacrifices._sacrificeCountByItemPersistentId[tag.GetString("persistentID")] = tag.GetInt("sacrificeCount");
-					if (ContentSamples.ItemNetIdsByPersistentIds.TryGetValue(tag.GetString("persistentID"), out int value2))
-						player.creativeTracker.ItemSacrifices.SacrificesCountByItemIdCache[value2] = tag.GetInt("sacrificeCount");
+				if (tag.ContainsKey("persistentID"))
+					continue; //Discard tags from previous insufficient implementation pre-alpha so they are not carried over to unloadedResearch
```

Comment:
Same thing without referencing the past.
```suggestion
				if (!tag.ContainsKey("mod") || !tag.ContainsKey("name"))
					continue; // Discard tags from previous insufficient implementation pre-alpha so they are not carried over to unloadedResearch
```